tools/functional/cli/test_cli_v4.py

```diff
@@ -561,11 +561,10 @@ def test_create_project_with_values_from_file(gitlab_cli, tmpdir):
     assert description in ret.stdout
 
 
-def test_create_project_deploy_token(gitlab_cli, project):
+def do_test_create_project_deploy_token(gitlab_cli, project, scopes, expected_scopes):
```

Comment:
hint: you could keep a single test function here and parametrize the input/expected output with pytest:

```python
@pytest.mark.parametrize(
    "scopes,expected_scopes",
    [
        ("read_registry", "['read_registry']"),
        ("read_registry,read_repository", "['read_repository', 'read_registry']"),
    ],
)
def test_create_project_deploy_token(gitlab_cli, project, scopes, expected_scopes):
    ...
```

Then you don't need a custom helper function at all and it will be easy to add more edge cases later. See very similar example here:
https://github.com/python-gitlab/python-gitlab/blob/af781c10db3829163f977e494e4008acf2096d64/gitlab/tests/test_config.py#L236-L249